src/TimeSeries.svelte

```diff
@@ -490,6 +492,111 @@
+              if (nextIndex == -1) {
+                datesToAdd.push({
+                  date: nextDate.toISOString().slice(0, 10),
+                  value: NaN
+                })
+              }
+            }
+          });
+          v.push(...datesToAdd)
+          v.sort((a,b) => new Date(a.date).getTime() - new Date(b.date).getTime());
+        });
+      });
+    }
+    */
+
+    /*  A function to add NaNs to the data in order to force breaks in the time series
+    *   when there are gaps in the weeks.
+    */ 
+   /*
+    function addNans(data) {
+      data.forEach((trend) => {
+        const trends = trend.trends;
+        Object.entries(trends).forEach(([k, v]) =>  {
+          const datesLength = v.length;
+          let datesToAdd: TrendValue[] = [];
+          v.forEach((trendVal: TrendValue, currentIndex: number) => {
+            const thisDate = trendVal.date;
+            let tempDate = convertStorageDate(thisDate);
+            // const nextExpectedDateString = new Date(thisDate.setUTCDate(thisDate.getUTCDate() + 7)).toUTCString();
+            // let nextExpectedDate = convertStorageDate(nextExpectedDateString);
+            let nextExpectedDate = new Date(tempDate.setUTCDate(tempDate.getUTCDate() + 7));
+            const nextExpectedDateString = nextExpectedDate.toISOString().slice(0, 10);
+            if (currentIndex < (datesLength-1)) {
+              const nextActualDate = v[currentIndex+1].date
+              if (nextActualDate != nextExpectedDateString) {
+                datesToAdd.push({
+                  date: nextExpectedDateString,
+                  value: NaN
+                })
+              }
+            }
+          });
+          v.push(...datesToAdd)
+          v.sort((a,b) => new Date(a.date).getTime() - new Date(b.date).getTime());
+        });
+      });
+    }
+    */
+    function addNans(data) {
+      console.log(data)
+      //console.log($dateRange)
+      const allDatesSet = new Set($dateRange);
+      //console.log(allDatesSet);
+      data.forEach((trend) => {
+        const trends = trend.trends;
+        const trendVals = Object.values(trends);
+        //console.log("trend values are:")
+        //console.log(trendVals);
+        trendVals.forEach((trendVal: any[]) => {
+          let datesToAdd: TrendValue[] = [];
+          const dateVals = trendVal.map((vals) => vals.date);
+          //console.log(dateVals);
+          const dateSet = new Set(dateVals);
+          //console.log(dateSet);
+          let missingDates = new Set([...allDatesSet].filter(x => !dateSet.has(x)));
+          //console.log(missingDates);
+          missingDates.forEach((missingDate) => {
+            datesToAdd.push({
+              date: missingDate,
+              value: NaN
+            })
+          });
+          trendVal.push(...datesToAdd)
+          trendVal.sort((a: TrendValue,b: TrendValue) => new Date(a.date).getTime() - new Date(b.date).getTime());
```

Comment:
You can compare the strings without any conversions.
We use the YYYY-mm-dd format so it's sortable as strings.

Reply:
using string comparison. let me know if this comparison function is ok.